ProcessLib/ThermalTwoPhaseFlowWithPP/CreateThermalTwoPhaseFlowWithPPProcess.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+#include "CreateThermalTwoPhaseFlowWithPPProcess.h"
+#include <cassert>
+#include "BaseLib/Functional.h"
+#include "MeshLib/MeshGenerators/MeshGenerator.h"
+#include "ProcessLib/Parameter/ConstantParameter.h"
+#include "ProcessLib/ThermalTwoPhaseFlowWithPP/CreateThermalTwoPhaseFlowWithPPMaterialProperties.h"
+#include "ProcessLib/ThermalTwoPhaseFlowWithPP/ThermalTwoPhaseFlowWithPPMaterialProperties.h"
+#include "ProcessLib/Utils/ParseSecondaryVariables.h"
+#include "ProcessLib/Utils/ProcessUtils.h"
+#include "ThermalTwoPhaseFlowWithPPProcess.h"
+#include "ThermalTwoPhaseFlowWithPPProcessData.h"
+
+namespace ProcessLib
+{
+namespace ThermalTwoPhaseFlowWithPP
+{
+std::unique_ptr<Process> createThermalTwoPhaseFlowWithPPProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config,
+    std::map<std::string,
+             std::unique_ptr<MathLib::PiecewiseLinearInterpolation>> const&
+        curves)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "THERMAL_TWOPHASE_WITH_PP");
+
+    DBUG("Create nonisothermal two-phase flow model.");
+    //! \ogs_file_param{prj__processes__process__TWOPHASE_FLOW_PRHO__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+
+    auto process_variables = findProcessVariables(
+        variables, pv_config,
+        {//! \ogs_file_param_special{prj__processes__process__THERMAL_TWOPHASE_FLOW_PP__process_variables__gas_pressure}
+         "gas_pressure",
+         //! \ogs_file_param_special{prj__processes__process__THERMAL_TWOPHASE_FLOW_PP__process_variables__capillary_pressure}
+         "capillary_pressure",
+         //! \ogs_file_param_special{prj__processes__process__THERMAL_TWOPHASE_FLOW_PP__process_variables__temperature}
+         "temperature"});
+
+    SecondaryVariableCollection secondary_variables;
+
+    NumLib::NamedFunctionCaller named_function_caller(
+        {"TwoPhaseFlow_pressure"});
+
+    ProcessLib::parseSecondaryVariables(config, secondary_variables,
+                                        named_function_caller);
+    // Specific body force
```

Comment:
Regarding this, I have discussed with Wenqing, since "specific_body_force " and "mass_lumping" are not special for two-phase processes, actually, it can further be shared by other processes.
so I would keep this untouched for now